### PR TITLE
playbook dispatcher book keeping

### PR DESCRIPTION
**Pull request by maxamillion**
This is the data model and initial book keeping mechanism for scheduling
and queuing playbook dispatcher jobs persistently.

Signed-off-by: Adam Miller <admiller@redhat.com>

### CODE REVIEW

File: pkg/updates/updates.go [hunk 1]
```diff
@@ -170,9 +172,20 @@ func updateFromHTTP(w http.ResponseWriter, r *http.Request) (*models.UpdateTrans
...
-		inventoryHosts = append(inventoryHosts, *updateDevice)
-		log.Debugf("updateFromHTTP::inventoryHosts: %#v", inventoryHosts)
-		update.InventoryHosts = inventoryHosts
+		devices = append(devices, *updateDevice)
+		log.Debugf("updateFromHTTP::devices: %#v", devices)
+		update.Devices = devices
+		log.Debugf("updateFromHTTP::update.Devices: %#v", devices)
+
+		dispatchRecord := new(models.DispatchRecord)
+		dispatchRecord.Device = updateDevice
```
Comment (acosferreira) [inline pkg/updates/updates.go:181]: Here you can use the repoURL from the upload

File: pkg/updates/updates.go [hunk 2]
```diff
@@ -170,9 +172,20 @@ func updateFromHTTP(w http.ResponseWriter, r *http.Request) (*models.UpdateTrans
...
 		log.Debugf("updateFromHTTP::updateDevice: %#v", updateDevice)
-		inventoryHosts = append(inventoryHosts, *updateDevice)
-		log.Debugf("updateFromHTTP::inventoryHosts: %#v", inventoryHosts)
-		update.InventoryHosts = inventoryHosts
+		devices = append(devices, *updateDevice)
+		log.Debugf("updateFromHTTP::devices: %#v", devices)
+		update.Devices = devices
+		log.Debugf("updateFromHTTP::update.Devices: %#v", devices)
+
+		dispatchRecord := new(models.DispatchRecord)
```
Comment (biancarosa) [inline pkg/updates/updates.go:180]: We actually do this like...:
```
&models.DispatchRecord{
      Device: updateDevice,
}
```
Makes it a little bit more easier to read, I guess. Just a suggestion though.

Comment (biancarosa) [reply pkg/updates/updates.go:180]: (with all of the fields, I didn't copy them all because I was lazy)

Comment (maxamillion) [reply pkg/updates/updates.go:180]: I like that, thanks!

**Review by biancarosa — APPROVED** [review]

**Review by acosferreira — APPROVED** [review]

**Pull request merged**